test/unit/util/test_cuda_vector_matrix.cpp

```diff
@@ -0,0 +1,187 @@
+#include <gtest/gtest.h>
+
+#include <micm/util/cuda_vector_matrix.hpp>
+
+#include "test_matrix_policy.hpp"
+
+template<class T>
+using Group1MatrixAlias = micm::CudaVectorMatrix<T,1>;
+template<class T>
+using Group2MatrixAlias = micm::CudaVectorMatrix<T,2>;
+template<class T>
+using Group3MatrixAlias = micm::CudaVectorMatrix<T,3>;
+template<class T>
+using Group4MatrixAlias = micm::CudaVectorMatrix<T,4>;
+
+
+
+TEST(VectorMatrix, SmallVectorMatrix)
+{
+  auto matrix = testSmallMatrix<Group2MatrixAlias>();
+
+  matrix.CopyToDevice();
+  matrix.GetFromDevice();
+
+  EXPECT_EQ(matrix[1][3], 64.7);
+  EXPECT_EQ(matrix[0][0], 41.2);
+  EXPECT_EQ(matrix[2][4], 102.3);
+
+  std::vector<double>& data = matrix.AsVector();
+
+  EXPECT_EQ(data.size(), 4 * 5);
+  EXPECT_EQ(matrix.GroupSize(), 2 * 5);
+  EXPECT_EQ(matrix.NumberOfGroups(), 2);
+  EXPECT_EQ(matrix.GroupVectorSize(), 2);
+  EXPECT_EQ(data[0], 41.2);
+  EXPECT_EQ(data[2 * 5 + 0 + 2 * 4], 102.3);
+  EXPECT_EQ(data[1 + 2 * 3], 64.7);
+}
+
+TEST(VectorMatrix, SmallConstVectorMatrix)
+{
+  auto matrix = testSmallConstMatrix<Group4MatrixAlias>();
+
+  const std::vector<double>& data = matrix.AsVector();
+
+  EXPECT_EQ(data.size(), 4 * 5);
+  EXPECT_EQ(matrix.GroupSize(), 4 * 5);
+  EXPECT_EQ(matrix.NumberOfGroups(), 1);
+  EXPECT_EQ(matrix.GroupVectorSize(), 4);
+  EXPECT_EQ(data[0], 41.2);
+  EXPECT_EQ(data[2 + 4 * 4], 102.3);
+  EXPECT_EQ(data[1 + 4 * 3], 64.7);
+}
```

Comment:
What is this test for?

Reply:
Ah, good catch!  Glossed over this one initially.  There's supposed to be copies to and from the device and verifying the contents afterwards. similar to the test above.